src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -111,10 +119,34 @@ class HttpGoogleCloudStorageDAO(
     storage.buckets().insert(projectId, bucket).execute()
   }
 
-  override def deleteBucket(ownerId: String, projectId: String, bucketName: String): Unit = {
-    val credential = getBucketCredential(ownerId)
-    val storage = getStorage(credential)
-    storage.buckets().delete(bucketName).execute()
+  override def deleteBucket(userId: String, projectId: String, bucketName: String): Unit = {
+    import system.dispatcher
+    def bucketDeleteFn(): Unit = {
+      val credential = getBucketCredential(userId)
+      val storage = new Storage.Builder(httpTransport, jsonFactory, credential).build()
+      val bucket: Bucket = storage.buckets().get(bucketName).execute()
+
+      //Google doesn't let you delete buckets that are full.
+      //You can either remove all the objects manually, or you can set up lifecycle management on the bucket.
+      //This can be used to auto-delete all objects next time the Google lifecycle manager runs (~every 24h).
+      //More info: http://bit.ly/1WCYhhf and http://bit.ly/1Py6b6O
+      val deleteEverythingRule = new Lifecycle.Rule()
+        .setAction(new Action().setType("delete"))
+        .setCondition(new Condition().setAge(0))
+
+      val lifecycle = new Lifecycle().setRule(List(deleteEverythingRule))
+      bucket.setLifecycle(lifecycle)
+
+      Try(storage.buckets().delete(bucketName).execute) match {
+        //Google returns 409 Conflict if the bucket isn't empty.
+        case Failure(gjre: GoogleJsonResponseException) if gjre.getDetails.getCode == 409 =>
+          system.scheduler.scheduleOnce(deletedBucketCheckSeconds seconds)(bucketDeleteFn)
```

Comment:
So is the upshot here that we attempt to delete the bucket every ```deletedBucketCheckSeconds```, until it finally succeeds?